drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java

```diff
@@ -0,0 +1,178 @@
+
+    private SubHeaderBuilder subHeaderBuilder;
+    private ColumnContext columnContext;
+    private final GuidedDecisionTable52 dtable;
+    private final Map<String, String> varListInOrder = new HashMap<>();// original, new one
+
+    public BRLColumnSubHeaderBuilderDefault(final SubHeaderBuilder subHeaderBuilder,
+                                            final ColumnContext columnContext,
+                                            final GuidedDecisionTable52 dtable) {
+        this.subHeaderBuilder = subHeaderBuilder;
+        this.columnContext = columnContext;
+        this.dtable = dtable;
+    }
+
+    @Override
+    public void brlActions(final BRLActionColumn brlColumn) {
+
+        final GuidedDecisionTable52 dt = new GuidedDecisionTable52();
+        dt.getAttributeCols().addAll(dtable.getAttributeCols());
+        dt.getMetadataCols().addAll(dtable.getMetadataCols());
+        dt.getConditions().addAll(dtable.getConditions());
+        final ArrayList<DTCellValue52> list = new ArrayList<>();
+
+        for (Object o : brlColumn.getDefinition()) {
+            if (o instanceof ActionInsertFact) {
+                ActionInsertFact insertFact = (ActionInsertFact) o;
+                if (insertFact.getBoundName() == null) {
+                    insertFact.setBoundName(getBoundName(insertFact));
+                }
+            }
+        }
+
+        for (int i = 0; i < dt.getExpandedColumns().size(); i++) {
+            list.add(dtable.getData().get(0).get(i));
+        }
+
+        list.addAll(setUpVarNamesWithTemps(brlColumn.getChildColumns()));
+        dt.getData().add(list);
+
+        dt.getActionCols().add(brlColumn);
+
+        subHeaderBuilder.addHeaderAndTitle(SubHeaderBuilder.ACTION,
+                                           brlColumn.getHeader());
+        subHeaderBuilder.getFieldRow().createCell(subHeaderBuilder.getTargetColumnIndex()).setCellValue(replaceTempVars(brlColumn.getChildColumns(),
+                                                                                                                        subString(GuidedDTDRLPersistence.getInstance().marshal(dt), "then", "end")));
+    }
+
+    private String getBoundName(final ActionInsertFact insertFact) {
+        if (hasContent(insertFact.getBoundName())) {
+            return insertFact.getBoundName();
+        } else {
+            return columnContext.getNextFreeColumnFactName();
+        }
+    }
+
+    @Override
+    public void brlConditions(final BRLConditionColumn brlColumn) {
+
+        for (IPattern iPattern : brlColumn.getDefinition()) {
+            if (iPattern instanceof FactPattern) {
+                columnContext.addBoundName(((FactPattern) iPattern).getBoundName());
+            }
+        }
+
+        final GuidedDecisionTable52 dt = new GuidedDecisionTable52();
+        dt.getConditions().add(brlColumn);
+        final ArrayList<DTCellValue52> list = new ArrayList<>();
+        list.add(new DTCellValue52(1));
```

Comment:
Is 1 a specific value? Maybe, using a constant can help to understand its purpose